rust/datafusion/src/physical_plan/hash_aggregate.rs

```diff
@@ -322,48 +325,76 @@ fn group_aggregate_batch(
             });
     }
 
+    // Collect all indices + offsets based on keys in this vec
+    let mut batch_indices: UInt32Builder = UInt32Builder::new(0);
```

Comment:
I wonder if you could get any additional performance by using the knowledge of the size of `batch_keys`

```suggestion
    let mut batch_indices: UInt32Builder = UInt32Builder::new(batch_keys.len());
```

Reply:
Or maybe you have to scale it by the number of accumulators too